src/lib/workflow-validator.ts

```diff
@@ -15,29 +15,39 @@
  *
  */
 
-import { DefinedError, ValidateFunction } from 'ajv';
+import { ValidateFunction } from 'ajv';
 import { Specification } from './definitions';
 import { validators } from './validators';
 
 export class WorkflowValidator {
   /** The validation errors after running validate(), if any */
-  errors: DefinedError[] | never[] = [];
+  errors: ValidationError[] | never[] = [];
   /** The validate function */
   private validateFn: ValidateFunction<Specification.Workflow>;
+
   /**
    * Creates a new WorkflowValidator for the provided workflow
    * @param {Workflow} workflow The workflow to validate
    */
   constructor(private workflow: Specification.Workflow) {
     this.validateFn = validators.get('Workflow') as ValidateFunction<Specification.Workflow>;
   }
+
   /**
    * Validates the workflow, populates the errors if any
    * @returns {boolean} If the workflow is valid or not
    */
   validate(): boolean {
     const isValid = this.validateFn(this.workflow);
-    this.errors = this.validateFn.errors as DefinedError[];
+    if (this.validateFn.errors) {
+      this.errors = this.validateFn.errors.map(
+        (error) => new ValidationError(`invalid: ${error.instancePath} | ${error.schemaPath} | ${error.message}`)
+      );
+    }
     return isValid;
   }
```

Comment:
The whole validate function can be removed and the validation can be done in the constructor using [utils.validate](https://github.com/serverlessworkflow/sdk-typescript/blob/main/src/lib/utils.ts#L10) as partially suggested in @antmendoza [comment](https://github.com/serverlessworkflow/sdk-typescript/pull/112#discussion_r642399235).

Reply:
I prefer to keep it if you don't mind and make it private. 

I am adding also an `isValid` property, 

should `errors`, and `isValid` be methods instead ? In java it would be, in ts idk..

Reply:
I was thinking about this:

```typescript
import { DefinedError } from 'ajv';
export class ValidationError {
  readonly message: string;
  constructor(readonly error: DefinedError) {
    this.message = `invalid: ${error.instancePath} | ${error.schemaPath} | ${error.message}`;
  }
}
```

Reply:
You can keep the method private indeed.

For `errors` and `isValid`, I don't understand why you'd want methods?

Reply:
I am just asking, exposing properties make the object mutable. 

I will keep as it is now.

Reply:
You can use getter only properties if you want to prevent someone to write it later on, outside of the class:
```typescript
private _isValid: boolean;
get isValid(): boolean {
  return this._isValid;
}
private validate() {
  this._isValid = false;
}
```

Reply:
I just realice that we can have them readonly if we set the value in the constructor.

Reply:
Yes and "readonly" is only at compile time, meaning that TS will throw an error when compiling, but, it doesn't make the property readonly at runtime in the end. (e.g.: https://bit.ly/3p5Nf8k )